src/Plugin/Field/FieldFormatter/ReactParagraphs.php

```diff
@@ -38,36 +44,39 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
       }
       $settings = json_decode($item['settings'], TRUE);
 
-      foreach ($elements as $key => $element) {
-        if ($item['target_id'] == $element['#paragraph']->id()) {
-          unset($elements[$key]);
-
-          $organized_elements[$settings['row']]['attributes'] = new Attribute(['class' => ['react-paragraphs-row']]);
-          $bundle = $element['#paragraph']->bundle();
-          $organized_elements[$settings['row']]['items'][$settings['index']] = [
-            'entity' => $element,
-            'width' => $settings['width'],
-            'attributes' => new Attribute(['class' => ['paragraph-item', Html::cleanCssIdentifier("ptype-$bundle")], 'data-react-columns' => $settings['width']]),
-          ];
-
-          if (!isset($organized_elements[$settings['row']]['width'])) {
-            $organized_elements[$settings['row']]['width'] = 0;
-          }
-          $organized_elements[$settings['row']]['width'] += $settings['width'];
-        }
+      // Always add attributes to the existing row.
+      $organized_elements[$settings['row']]['attributes'] = new Attribute(['class' => ['react-paragraphs-row']]);
+
+      $bundle = $elements[$delta]['#paragraph']->bundle();
+
+      // Add the paragraph render array to the list of items in the correct row.
+      $organized_elements[$settings['row']]['items'][$settings['index']] = [
+        'entity' => $elements[$delta],
+        'width' => $settings['width'],
+        'attributes' => new Attribute(['class' => ['paragraph-item', Html::cleanCssIdentifier("ptype-$bundle")], 'data-react-columns' => $settings['width']]),
+      ];
+
+      // Add up the width of all elements with each row to provide a spacer in
+      // the later steps.
+      if (!isset($organized_elements[$settings['row']]['width'])) {
+        $organized_elements[$settings['row']]['width'] = 0;
       }
+      $organized_elements[$settings['row']]['width'] += $settings['width'];
     }
 
+    // Ensure we have all the rows and items sorted correctly based on their row
+    // and item indexes.
     ksort($organized_elements);
     array_walk($organized_elements, 'ksort');
 
+    // Add any spacers to the end of the rows.
     $this->addSpacers($organized_elements);
 
+    // Return a field render array. Set `#is_multiple` to false to reduce
+    // unnecessary markup since the included template handles the multiple.
     return [
-      [
-        '#theme' => 'react_paragraphs',
-        '#rows' => $organized_elements,
-      ],
+      ['#theme' => 'react_paragraphs', '#rows' => $organized_elements],
+      '#is_multiple' => FALSE,
```

Comment:
🤔 This might be a gotcha later. Not sure what the long term ramifications of this would be.

Reply:
So all this does in the end is reduce the markup by 1 div when using templates from core. I don't forsee much issues in the future with it. preprocess_field can always modify or we can always remove the line and restore the extra div.

https://git.drupalcode.org/project/drupal/blob/8.8.x/core/themes/stable/templates/field/field.html.twig#L46

Reply:
Theming and hooks depend on this value so expectations and outcomes may differ but that is something we can tackle later.